src/Http/Http.Abstractions/src/HttpResponse.cs

```diff
@@ -111,5 +113,15 @@ public abstract class HttpResponse
         /// where only ASCII characters are allowed.</param>
         /// <param name="permanent"><c>True</c> if the redirect is permanent (301), otherwise <c>false</c> (302).</param>
         public abstract void Redirect(string location, bool permanent);
+
+        /// <summary>
+        /// Starts the response by calling OnStarting() and making headers unmodifiable.
+        /// </summary>
+        /// <param name="flush"><c>True</c> if headers and request line should be flushed, otherwise <c>false</c>.</param>
+        /// <param name="cancellationToken"></param>
+        /// <remarks>
+        /// If the <see cref="IHttpResponseStartFeature"/> isn't set, StartAsync will default to calling HttpResponse.Body.FlushAsync().
+        /// </remarks>
+        public abstract Task StartAsync(bool flush, CancellationToken cancellationToken = default);
```

Comment:
If the only reason to call HttpResponse.StartAsync() is to make the BodyPipe usable, this method should return the PipeWriter and replace the BodyPipe property. You could call it InitialzeBodyPipeAsync() or something.

That way, your first indication that you need to call a method before using the response body PipeWriter won't be a runtime exception.

Reply:
Why include a flush parameter? Couldn't you always to an empty flush immediately after calling this method if you cared to do so?

Reply:
We need the API for pipes but there's nothing in the implementation that ties it to pipes. Coupling it to pipes will also make it harder to wrap.

Reply:
If you aren't using pipes, it would be far more idiomatic to call `HttpContext.Response.Body.FlushAsync()`. This feature is very much tied to pipes whether or not the implementation needs them.

Any usage for this feature that doesn't involve pipes is purely academic. We might as well use a name that people looking to use pipes will actually find in intellisense.

Reply:
We're not tying it to pipes. We can discuss it tomorrow.